src/wolfgame.py

```diff
@@ -6707,24 +6707,30 @@ def listroles(cli, nick, chan, rest):
     elif rest[0] and not rest[0].isdigit():
         gamemode = rest[0]
         if gamemode not in var.GAME_MODES.keys():
-            gamemode, _ = complete_match(rest[0], var.GAME_MODES.keys() - ["roles", "villagergame"] - var.DISABLED_GAMEMODES)
-        validMode = gamemode in var.GAME_MODES.keys() and gamemode != "roles" and gamemode != "villagergame" and gamemode not in var.DISABLED_GAMEMODES
-        if validMode and hasattr(var.GAME_MODES[gamemode][0](), "ROLE_GUIDE"):
-            mode = var.GAME_MODES[gamemode][0]()
+            gamemode = complete_match(rest[0], var.GAME_MODES.keys() - ["roles", "villagergame"] - var.DISABLED_GAMEMODES)
+            if len(gamemode) > 1:
+                msg.append("{0}: {1} is not a valid game mode. Possbile modes are: {2} ".format(nick, rest[0], ", ".join(gamemode)))
+                rest = []
+                roleindex = {}
+            else:
+                gamemode = gamemode.pop()
+        else:
+            validMode = gamemode in var.GAME_MODES.keys() and gamemode != "roles" and gamemode != "villagergame" and gamemode not in var.DISABLED_GAMEMODES
```

Comment:
this is fugly, use set union to check for disabled gamemodes, roles, and villagergame in one operation. I know this wasn't you but since you're touching the code anyway, you might as well fix this up